executor/builder.go

```diff
@@ -433,7 +433,7 @@ func buildIndexLookUpChecker(b *executorBuilder, p *plannercore.PhysicalIndexLoo
 
 	tps := make([]*types.FieldType, 0, fullColLen)
 	for _, col := range is.Columns {
-		tps = append(tps, &(col.FieldType))
+		tps = append(tps, col.FieldType.ArrayType())
```

Comment:
Why is it `ArrayType()`?

Reply:
the index data is encoded with the element type in the array, so we should read the index in the same type

Reply:
But for the other types, the clone is unnecessary and it is confusing.

Reply:
+1, this seems confusing for me. Need some comments maybe.